Generic Whitespace Check, updated method references processing, issue #6...

alexkravin opened the pull request:
...77

According to #677 

Fixed false-positive in case if generic's end follows with whitespace after method reference, e.g.:

```java
final Supplier<Optional<Integer>> function = Optional::<Integer>empty;
```

Also updated the Check so it warns if in such case generic will be followed by whitespace:

```java
final Supplier<Optional<Integer>> function = Optional::<Integer> empty; // warn here
```

romani commented:
1) please update javadoc to show example of method ref processing - https://github.com/alexkravin/checkstyle/blob/adfe1d1794745a624ee2e5e892e12698a273655b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java#L63
that same for xdoc.

2) please confirm that after fix google-style-config-test is passed and generation of validation report against Guava is the same.

alexkravin commented:
1) Updated javadoc/xdoc

2) Google-style-test passes with success, reports against Guava are the same

romani commented:
Please make an example of method reference as minimal as possible to let it fit a single line.
Please review config_whitespace.html , code example should looks pretty.

alexkravin commented:
Updated, added examples with method references usage from previous and current issues.
Updated xdoc, looks ok
![config](https://cloud.githubusercontent.com/assets/6840443/6462944/c1d0aca0-c1be-11e4-9221-385ef04b0513.png)

romani commented:
Please allign comments to the right

alexkravin commented:
Updated javadoc

romani commented:
merged as FF
